zipline/data/bundles/core.py

```diff
@@ -321,53 +340,56 @@ def ingest(name,
         cachepath = cache_path(name, environ=environ)
         pth.ensure_directory(pth.data_path([name, timestr], environ=environ))
         pth.ensure_directory(cachepath)
-
         with dataframe_cache(cachepath, clean_on_failure=False) as cache, \
                 ExitStack() as stack:
             # we use `cleanup_on_failure=False` so that we don't purge the
             # cache directory if the load fails in the middle
-
             if bundle.create_writers:
-                daily_bars_path = stack.enter_context(working_dir(
-                    daily_equity_path(name, timestr, environ=environ),
-                )).path
+                wd = stack.enter_context(working_dir(
+                    pth.data_path([], environ=environ))
+                )
+                daily_bars_path = wd.mkdir(*daily_equity_relative(
+                    name, timestr, environ=environ)
+                )
                 daily_bar_writer = BcolzDailyBarWriter(
                     daily_bars_path,
                     bundle.calendar,
+                    nyse_cal
                 )
                 # Do an empty write to ensure that the daily ctables exist
                 # when we create the SQLiteAdjustmentWriter below. The
                 # SQLiteAdjustmentWriter needs to open the daily ctables so
                 # that it can compute the adjustment ratios for the dividends.
+
                 daily_bar_writer.write(())
                 minute_bar_writer = BcolzMinuteBarWriter(
                     bundle.calendar[0],
-                    stack.enter_context(working_dir(
-                        minute_equity_path(name, timestr, environ=environ),
-                    )).path,
+                    wd.mkdir(*minute_equity_relative(
+                        name, timestr, environ=environ)
+                    ),
                     bundle.opens,
                     bundle.closes,
                     minutes_per_day=bundle.minutes_per_day,
                 )
-                asset_db_writer = AssetDBWriter(
-                    stack.enter_context(working_file(
-                        asset_db_path(name, timestr, environ=environ),
-                    )).path,
-                )
-                adjustment_db_writer = SQLiteAdjustmentWriter(
-                    stack.enter_context(working_file(
-                        adjustment_db_path(name, timestr, environ=environ),
-                    )).path,
-                    BcolzDailyBarReader(daily_bars_path),
-                    bundle.calendar,
-                    overwrite=True,
+                asset_db_writer = stack.enter_context(
+                    AssetDBWriter(
+                        wd.getpath(*asset_db_relative(
+                            name, timestr, environ=environ)
```

Comment:
With 2 paren, it would look like this?
```python
AssetDBWriter(
    wd.getpath(*asset_db_relative(
        name, timestr, environ=environ,
    )),
)
```